### ISSUE TITLE
Mdx.playerdata.writeToJson won't serialize com.badlogic.gdx.utils.ObjectMap;

### ISSUE THREAD

**Issue by Khazrak** [mini2Dx/mini2Dx#37]
I have a class:

public class SaveGameData {
  @Field
  private ObjectMap<String, InGameScreen> mapDataObjectMap;
  /** more code */
}

The class InGameScreen has a couple of Strings annotated with @Field.

When I try  Mdx.playerData.writeJson(saveGameData, path);

The map is empty (I've put data in it and tested it to be sure that it was there).
(The json shows empty objects {} )

A normal HashMap can be serialized with the same data, but doesn't deserialized .
com.badlogic.gdx.utils.reflect.ReflectionException: Could not instantiate instance of class: se.codeslasher.legend.screens.InGameScreen
at com.badlogic.gdx.utils.reflect.ClassReflection.newInstance(ClassReflection.java:70)
	at org.mini2Dx.core.serialization.JsonSerializer.deserialize(JsonSerializer.java:187)
	at org.mini2Dx.core.serialization.JsonSerializer.setMapField(JsonSerializer.java:344)
	at org.mini2Dx.core.serialization.JsonSerializer.setField(JsonSerializer.java:269)
	at org.mini2Dx.core.serialization.JsonSerializer.deserialize(JsonSerializer.java:207)
	at org.mini2Dx.core.serialization.JsonSerializer.fromJson(JsonSerializer.java:45)
	at org.mini2Dx.desktop.playerdata.DesktopPlayerData.readJson(DesktopPlayerData.java:103)

Maybe implement support for custom serialization?
I really like the Mdx.playerdata so I hope this can be fixed (or another way of using it)

**Comment by tomcashman** [comment]
Can I ask what types the keys/values are for the map? HashMap should work unless there's some weird edge case missed in the [unit test](https://github.com/mini2Dx/mini2Dx/blob/master/core/src/test/java/org/mini2Dx/core/serialization/JsonSerializerTest.java).

I was unaware LibGDX had its own implementations of sets/maps/etc. - I'll schedule support for these in the next major release.

**Comment by Khazrak** [comment]
```java
public class SaveGameData {
    
    @Field
    private String PLACEHODER_METADATA = "Saved Game";
    @Field
    private int currentMapId = 3; //TODO: change to firstMap
    @Field
    private ObjectMap<String, InGameScreen> mapDataObjectMap;
    //private HashMap<String, InGameScreen> mapDataObjectMap;
    @Field
    private ObjectMap<String, InGameScreen> idMapDataObjectMap;
    //private Map <Integer, InGameScreen> idMapDataObjectMap;

    public SaveGameData() {
        mapDataObjectMap = new ObjectMap<String, InGameScreen>();
        idMapDataObjectMap = new ObjectMap<Integer,InGameScreen>();
        //mapDataObjectMap = new HashMap<String, InGameScreen>();
        //idMapDataObjectMap = new HashMap<Integer, InGameScreen>();
    }

    public void addMapData(InGameScreen mapData) {
        this.mapDataObjectMap.put(mapData.getMapName(),mapData);
        this.idMapDataObjectMap.put(mapData.getId(), mapData);
    }
....(Getters and Setters)
```

And InGameScreen is just:
```java
public abstract class InGameScreen extends BasicGameScreen {

    private SnapshotArray <Enemy> enemieArray;

    public InGameScreen()
    {

    }
    public abstract int getId();
    public abstract String getMapName();
}
```

It's very early code right now. The HashMap is commented out right now. Since ObjectMap seems to use the "toString" value of the key Integer keys doesn't work properly so I use IntMap instead (not in this code example).

**Comment by tomcashman** [comment]
Your InGameScreen class is abstract so Mdx.playerData doesn't know how to construct an instance of it during deserialisation. This is why HashMap isn't working.

However, I'll still need to add support for LibGDX collections, especially as you're using SnapshotArray.

**Comment by tomcashman** [comment]
This is now available in 1.3.0
